### DISCUSSION TITLE
How to use only Yubikey without Phone?

### DISCUSSION THREAD

**Discussion by zDEFz** [community/community#77994] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I have 5 Yubikey 5 NFC with HMAC enabled. I do not use the TOTP. 
How can I use only my Yubikey along with the password to authenticate?
Passkey didn't work...

Found similar thread, but doesn't answer my question. https://github.com/orgs/community/discussions/22500

Relevant part

    I want to not allow SMS for 2FA to my account,
    I want to not allow TOTP for 2FA to my account,
    I want to not allow GitHub Mobile for 2FA to my account,
    I want to not allow recovery codes for 2FA to my account, and
    I want to be locked out if I lose all ≥2 of my security keys.

**Comment by vablings** [comment]
It seems that for the rollout of 2FA, a security key is not considered.

Honestly I think its fucking bullshit that I am forced to use an inferior method for security.
There are options to use other apps for TOTP what can be secured with a security key but why should I have to.

**Comment by zDEFz** [comment]
> It seems that for the rollout of 2FA, a security key is not considered.
> 
> Honestly I think its fucking bullshit that I am forced to use an inferior method for security. There are options to use other apps for TOTP what can be secured with a security key but why should I have to.


Actually I ended up using `oathtool --totp --base32 "mysecretkey"` 
Afterwards I could add my Yubikey.
But no, this is not something Github explains to the user.
This is bad UX.

**Reply by vablings** [reply]
But then you have to use cli tools to fetch the topt key every time you want to log in? Rather than just plugging in the key and using fido2?

**Reply by zDEFz** [reply, marked as answer]
No. Just once. And then you'd save the recovery keys, including your secret key to a keepassxc database.
When you login, you login with your password and your yubikey, as long as it is added as security key.

But that other method, using oathtool is something you regularly wouldn't do when a security key is deployed...
It's kinda dumb when you think about it, isn't it?

**Comment by PyrateAkananto** [comment]
Please note that with the "Yubico Authenticator" app (in my case the Windows desktop app) you can complete do the whole TOTP process for the initial 2FA setup at GitHub and store everything related to it (except GitHub's recovery codes) on a Yubikey.

I did that myself and _then_ additionally added the Yubikey as a security key to my GitHub account.

To sign in to GitHub I can now _either_ insert and press the YubiKey _or_ I can insert the Yubikey, launch the Yubico Authenticator app, read the TOTP from the app and enter it on GitHub.

**Reply by zDEFz** [reply]
Can you please make a small video tutorial on it?

**Reply by PyrateAkananto** [reply]
Video: no. Text: yes. Here is what I did:

1. Log into the website of GitHub.
2. Go to [Settings > Password and authentication](https://github.com/settings/security).
3. Add an "Authenticator app" and proceed until the QR code is visible on the screen.
4. Minimize the browser (don't close it).
5. Install Yubico Authenticator for Windows.
6. Insert Yubikey 5 into an USB slot.
7. Click the settings button in the top right corner (there really is not much functionality in the app and everything is pretty self-explanatory).
8. Click Setup > Add account.
9. Now maximize the browser again and open the windowed Yubico Authenticator app _in front of it_ so you can see both the QR code and the "Scan QR code" button on the screen.
10. Click the "Scan QR code" button (it seems to take a screenshot and search for a QR code in it).
11. Enter some more details in the Yubico Authenticator app (nickname for the account and such) and probably click some "save" button or something like that (sorry, I don't remember exactly and cannot repeat the process now). Afterwards everything should be stored on the Yubikey.
12. From now on when the Yubico Authenticator app is open and the Yubikey is inserted you see a list of accounts stored on the Yubikey.
13. Click the account that was just stored. There now is a six-digit number with a little animated time-remaining-icon next to it. That six-digit code is a TOTP (timed one-time password).
14. Back in the browser on the GitHub website enter that six-digit number (do it fast because the TOTP expires).
15. Afterwards complete the process as instructed. *Save the recovery codes in a secure storage!*
16. Now an "Authenticator app" is added to the GitHub account and a TOTP will be required for every sign-in at GitHub.

Optionally you now can even add a "Security key" to GitHub, using the very same Yubikey. That process is even simpler and works if your browser supports WebAuthn, which every modern browser should do. After you even completed that second process, including setting "Preferred 2FA method" to "Security key" instead of "Authenticator app", you can use your Yubikey during sign-in at GitHub _without_ needing the Yubico Authenticator app or any TOTP. Only inserting and tapping the Yubikey is required.

And you can and should add all your Yubikeys as "Security keys" to GitHub in case you lose one. The whole first process (as described above) has to be done only once, for the first Yubikey, not all the others.

![github-authentication](https://github.com/community/community/assets/11444373/2d0f62f9-23d3-4f87-9b22-c1bcceac5953)
